tests/test_api.py

```diff
@@ -338,6 +338,37 @@ def test_properties(self):
         self.assertEqual(self.dbus_props.Get('org.freedesktop.Test.Other', 'color'),
                          'yellow')
 
+        changed_props = []
+        ml = GLib.MainLoop()
+
+        def catch(*args, **kwargs):
+            if kwargs['interface'] == 'org.freedesktop.DBus.Properties' and \
+               kwargs['member'] == 'PropertiesChanged':
+                [iface, changed, _invalidated] = args
+                if str(iface) == 'org.freedesktop.Test.Main':
+                    changed_props.append(changed)
+                    ml.quit()
+
+        self.dbus_con.add_signal_receiver(catch,
+                                          interface_keyword='interface',
+                                          path_keyword='path',
+                                          member_keyword='member')
+
+        # change property using mock helper
+        self.dbus_mock.UpdateProperties('org.freedesktop.Test.Main', {
+            'version': 5,
+            'connected': False,
+        })
+
+        GLib.timeout_add(3000, ml.quit)
+        ml.run()
+
+        self.assertEqual(self.dbus_props.GetAll('org.freedesktop.Test.Main'),
+                         {'version': 5, 'connected': False})
+        self.assertEqual(len(changed_props), 1)
+        self.assertEqual(changed_props[0],
+                         {'version': 5, 'connected': False})
```

Comment:
Please drop the first and compare the entire `changed_props` to `[{...}]`. With that style, assertion failures are much more useful, as they will actually show you the extra unexpected entries.

Reply:
Dropped the length check.